src/commands/test/lib/calabash-preparer.ts

```diff
@@ -64,28 +64,18 @@ export class CalabashPreparer {
     }
 
     if (this.testParameters && this.testParameters.length > 0) {
-      command += ` --test-params ${this.generateTestParameterArgs()}`;
+      command += this.generateTestParameterArgs();
     }
 
     return command;
   }
 
   private generateTestParameterArgs(): string {
     let result: string = "";
-
-    if (this.testParameters) {
-      this.testParameters.forEach(p => {
-        let parsedParameter = parseTestParameter(p);
-        if (result != "") {
-          result += ",";
-        }
-        result += `${parsedParameter.key}:`;
-        if (parsedParameter.value != null) {
-          result += `${parsedParameter.value}`;
-        }
-      });
-    }
-
+    this.testParameters.forEach(p => {
+      let parsedParameter = parseTestParameter(p);
+      result += ` --test-parameters "${parsedParameter.key}:${parsedParameter.value}"`;
```

Comment:
This the first example I have seen of passing multiple `--test-parameters` args to `test-could submit`.

The command did not fail when I passed multiple `--test-parameters`, but do we know the correct parameters are uploaded to Test Cloud?    Do the arguments clobber each other?  @krukow If you don't have time to check, I can.

I use this format for `--test-parameters`

```
--test-parameters "pipeline:my-new-feature inject:https://example.com/my.dylib app_env:EMAIL=moody@example.com"
```

Reply:
Wow. I've just checked this. Neither of the options work for the generate command. Yet, the whole invocation still works end to end! 

I'm digging more into this.

Reply:
OK. Found multiple issues. I'm updating this PR to fix and explain.

The correct format for `test-cloud prepare` is:
```
test-cloud prepare XTCAndroidSample.apk \
 --artifacts-dir ~/tmp/prepare \
--test-parameters "foo:bar" "baz:qux"
```

```
cat ~/tmp/prepare/manifest.json
{
  "schemaVersion": "1.0.0",
  "files": [
    "features",
    "vendor",
    "Gemfile",
    "Gemfile.lock",
    "test_servers/1653c4806487278f440ae335a8888eb9_0.8.2.apk"
  ],
  "testFramework": {
    "name": "calabash",
    "data": {
      "foo": "bar",
      "baz": "qux"
    }
  }
}
```